python/ray/reporter.py

```diff
@@ -157,6 +185,14 @@ def perform_iteration(self):
         )
 
     def run(self):
+        """Publish the port."""
+        thread_pool = futures.ThreadPoolExecutor(max_workers=10)
+        server = grpc.server(thread_pool, options=(("grpc.so_reuseport", 0), ))
+        reporter_pb2_grpc.add_ReporterServiceServicer_to_server(
+            ReporterServer(), server)
+        port = server.add_insecure_port("[::]:0")
+        server.start()
+        self.redis_client.set(self.ip, port)
```

Comment:
This should be replaced by something like `self.redis_client.set("REPORTER_PORT:" + self.ip, port)` and then similar for the `.get` call above, to make sure the key is unique and doesn't clash with anything else.